Adding Hungarian Algorithm

alemini18 opened the pull request:
Adding Hungarian Algorithm, translated from e-maxx.ru.

adamant-pwn commented on src/graph/hungarian-algorithm.md:
```diff
@@ -0,0 +1,291 @@
+
+- for all vertices $j\in Z_2$, do $v[j] \gets v[j]-\Delta$.
+
+Note that conditions are not mutually exclusive.
```
I'm sorry, which conditions not mutually exclusive with what?

alemini18 replied:
I think this sentence was from a deleted paragraph or similar. I deleted it.

adamant-pwn commented on src/graph/hungarian-algorithm.md:
```diff
@@ -0,0 +1,291 @@
+
+$$minv[j]=\min_{i\in Z_1} A[i][j]-u[i]-v[j].$$
+
+It's easy to see that the desired value $\Delta$ is expressed in terms of them as follows:
```
This, and a paragraph below should have the same tabulation as "- To quickly recalculate...", but it breaks because of block maths. It'd be better to fix this

alemini18 replied:
I added css to math blocks to fix tabulation. It's not so elegant but it works.

alemini18 commented:
I've added the section about successive shortest paths algorithm: to keep it short, I mentioned only the analogy between the potentials and the subgraphs. 
For the dual problem, I added a remark under the task example that refers to it.
